roles/jboss_ws/tasks/main.yml

```diff
@@ -39,52 +38,31 @@
   ignore_errors: yes
   when: 
     - 'jws_find_home is defined'
-      #- "'results' in jws_find_home"
-      #- "jws_find_home['results'] | length > 1"
-      #- "'stdout_lines' in jws_find_home['results'][0]"
-      #- "jws_find_home['results'][0]['stdout_lines'] | length > 1"
+    - "'results' in jws_find_home"
+    - "jws_find_home['results'] | length > 0"
+    - "'stdout_lines' in jws_find_home['results'][0]"
+    - "jws_find_home['results'][0]['stdout_lines'] | length > 0"
 
 # ------------------------------ DETECT JWS/EWS -------------------------------
 - name: check jws presence with yum
-  raw: yum grouplist jws3 jws3plus jws5 | grep -A1 'Installed Groups:' | grep 'Red Hat JBoss Web Server [0-99]' 2>/dev/null
-  register: jws_check_installed_with_rpm
+  raw: yum grouplist jws3 jws3plus jws5 2>/dev/null | grep -A1 'Installed Groups:' | grep 'Red Hat JBoss Web Server [0-99]' 
+  register: jws_installed_with_rpm
```

Comment:
Didn't catch this last time, but `yum` will require privilege escalation. You should add `become: True` so sudo, etc will be used.

Reply:
yum grouplist doesn't require sudo